### PR TITLE
Fix file oparetion fails on high load

### PR DESCRIPTION
**Pull request by sartor**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

There are more fixes like https://github.com/yiisoft/cache-file/pull/50
I've added simple test, but not on `phpunit`, because it needs `pcntl` and uses forks to simulate real high concurrency load
All fixed cases are real, taken form my corporate high load product error log

### CODE REVIEW

File: concurrency_test.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @author sartor
```
Comment (samdark) [inline concurrency_test.php:4]: The file should be moved to `tests`.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Need a line for CHANGELOG. Also, it it possible to convert the test to PHPUnit one so it runs in CI?

**Comment by sartor** [comment]
If I put this code to PHPUnit it behaves strange because of multiple forks. I can run this test to see it exit code.

**Comment by samdark** [comment]
Let's try it.

**Comment by sartor** [comment]
It was easier than i thought

File: src/FileCache.php [hunk 2]
```diff
@@ -450,4 +453,27 @@ private function iterableToArray(iterable $iterable): array
         /** @psalm-suppress RedundantCast */
         return $iterable instanceof Traversable ? iterator_to_array($iterable) : (array) $iterable;
     }
+
+    /**
+     * Check if error was because of file was already deleted by another process on high load
+     */
+    private function isLastErrorNotSafe(mixed $result): bool
```
Comment (vjik) [inline src/FileCache.php:460]:
```suggestion
    private function isLastErrorSafe(bool $result): bool
```

And use above, for example:

```php
return  $this->isLastErrorSafe(@touch($file, $expiration));
```

**Comment by samdark** [comment]
Still need a line for CHANGELOG.

File: src/FileCache.php [hunk 3]
```diff
@@ -450,4 +447,27 @@ private function iterableToArray(iterable $iterable): array
         /** @psalm-suppress RedundantCast */
         return $iterable instanceof Traversable ? iterator_to_array($iterable) : (array) $iterable;
     }
+
+    /**
+     * Check if error was because of file was already deleted by another process on high load
+     */
+    private function isLastErrorSafe(mixed $result): bool
```
Comment (vjik) [inline src/FileCache.php:454]:
```suggestion
    private function isLastErrorSafe(bool $result): bool
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ## 2.0.2 under development
 
 - Chg #50: Do not throw exception on file delete when file not found (fix for high concurrency load) (@sartor)
+- Chg #52: More highload errors on file delete fixed, separate concurrency test added (@sartor)
```
Comment (vjik) [inline CHANGELOG.md:6]:
```suggestion
- Chg #52: Do not throw exception on  change mode and modification time of cache file (fix for high
  concurrency load) (@sartor)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,8 @@
 ## 2.0.2 under development
 
 - Chg #50: Do not throw exception on file delete when file not found (fix for high concurrency load) (@sartor)
+- Chg #52: Do not throw exception on  change mode and modification time of cache file (fix for high
```
Comment (vjik) [inline CHANGELOG.md:6]:
```suggestion
- Chg #52: Do not throw exception on change mode and modification time of cache file (fix for high
```

**Review by vjik — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**